Feature: Add the makeAllSearchableUsing API to the default import source

yocmen opened the pull request:
In this PR I'm adding the handle of the method `makeAllSearchableUsing` in the `DefaultImportSource`to allow/match this to the scout docs.

https://laravel.com/docs/10.x/scout#modifying-the-import-query

I did this because I found a case where I needed to use the `makeAllSearchableUsing` method in a Model and found this was the easiest way to do it.

hkulekci commented:
this means that for each query this method will work, right? What do you think, adding it with a criteria instead of applying for each query?

yocmen commented:
Hi @hkulekci, the Searchable Trait already have the `makeAllSearchableUsing` method just returning the query. The implementation is identical to the scout one (using when(true..) so any searchable model will have that method by default and ready to overwrite it in the model with new logic when you need it, I can add a "exists method" check if you want it, I just did it like the scout implementation and is working really good for me.

Thanks, and let me know anything :)

yocmen commented:
hi @matchish any thoughts on this? :)

matchish commented:
Hey) thanks for your effort to create the PR. Small improvement that should improve readability
```php
        $query = $this->className::makeAllSearchableUsing($this->model()->newQuery());
        $softDelete = $this->className::usesSoftDelete() && config('scout.soft_delete', false);
        $query
            ->when($softDelete, function ($query) {
                return $query->withTrashed();
            })
            ->orderBy($this->model()->getQualifiedKeyName());
        $scopes = $this->scopes;
```
And could you add a test to cover the PR?

yocmen commented:
Hi @matchish, I added a new Test, dont know if that was the right place, let me know.

In order to add the Test I added a new `Post` model, including migration and factory, on this model I added the `makeAllSearchableUsing` method modifing the query to be used when the data is imported.

This was the best way I found to make a test, I can't figure how to mock any existing model to assert if the method was called, so instead I added like this.

Please let me know anything, all help is welcome.

yocmen commented:
Hi @matchish @hkulekci what do you think on my last changes?
